Assets/MRTK/StandardAssets/Shaders/MixedRealityTextMeshProSprite.shader

```diff
@@ -0,0 +1,194 @@
+
+// NOTE: MRTK Shaders are versioned via the MRTK.Shaders.sentinel file.
+// When making changes to any shader's source file, the value in the sentinel _must_ be incremented.
+
+// Text Mesh Pro Sprite shader with MRTK Additions
+
+// MRTK Additions
+// - Single Pass Instanced Stereo Rendering Support
+// - Support for Clipping Primitives (Plane, Sphere, Box)
+// - Added to MRTK namespace
+
+Shader "Mixed Reality Toolkit/TextMeshProSprite"
+{
+	Properties
+	{
+		_MainTex ("Sprite Texture", 2D) = "white" {}
+		_Color ("Tint", Color) = (1,1,1,1)
+		
+		_StencilComp ("Stencil Comparison", Float) = 8
+		_Stencil ("Stencil ID", Float) = 0
+		_StencilOp ("Stencil Operation", Float) = 0
+		_StencilWriteMask ("Stencil Write Mask", Float) = 255
+		_StencilReadMask ("Stencil Read Mask", Float) = 255
+
+		_ColorMask ("Color Mask", Float) = 15
+		_ClipRect ("Clip Rect", vector) = (-32767, -32767, 32767, 32767)
+
+		[Toggle(UNITY_UI_ALPHACLIP)] _UseUIAlphaClip ("Use Alpha Clip", Float) = 0
+	}
+
+	SubShader
+	{
+		Tags
+		{ 
+			"Queue"="Transparent" 
+			"IgnoreProjector"="True" 
+			"RenderType"="Transparent" 
+			"PreviewType"="Plane"
+			"CanUseSpriteAtlas"="True"
+		}
+		
+		Stencil
+		{
+			Ref [_Stencil]
+			Comp [_StencilComp]
+			Pass [_StencilOp] 
+			ReadMask [_StencilReadMask]
+			WriteMask [_StencilWriteMask]
+		}
+
+		Cull Off
+		Lighting Off
+		ZWrite Off
+		ZTest [unity_GUIZTestMode]
+		Blend SrcAlpha OneMinusSrcAlpha
+		ColorMask [_ColorMask]
+
+		Pass
+		{
+		CGPROGRAM
+			#pragma vertex vert
+			#pragma fragment frag
+
+			#pragma multi_compile __ UNITY_UI_CLIP_RECT
+                        #pragma multi_compile __ UNITY_UI_ALPHACLIP
+
+                        #pragma multi_compile __ _CLIPPING_PLANE _CLIPPING_SPHERE _CLIPPING_BOX
+
+                        #include "UnityCG.cginc"
+			#include "UnityUI.cginc"
+			#include "MixedRealityShaderUtils.cginc"
+
+#if defined(_CLIPPING_PLANE) || defined(_CLIPPING_SPHERE) || defined(_CLIPPING_BOX)
+        #define _CLIPPING_PRIMITIVE
+#else
+        #undef _CLIPPING_PRIMITIVE
+#endif
+
+#if defined(_CLIPPING_PLANE)
+        fixed _ClipPlaneSide;
+        float4 _ClipPlane;
+#endif
+
+#if defined(_CLIPPING_SPHERE)
+        fixed _ClipSphereSide;
+        float4x4 _ClipSphereInverseTransform;
+#endif
+
+#if defined(_CLIPPING_BOX)
+        fixed _ClipBoxSide;
+        float4x4 _ClipBoxInverseTransform;
+#endif
+
+        uniform float        _VertexOffsetX;
+        uniform float        _VertexOffsetY;
+
+			struct appdata_t
+			{
+                float4 vertex   : POSITION;
+				float4 color    : COLOR;
+				float2 texcoord : TEXCOORD0;
+                UNITY_VERTEX_INPUT_INSTANCE_ID
+			};
+
+			struct v2f
+			{
+                
+                float4 vertex   : SV_POSITION;
+				fixed4 color    : COLOR;
+				half2 texcoord  : TEXCOORD0;				
+#if defined(_CLIPPING_PRIMITIVE)
+                float3 worldPosition    : TEXCOORD1;
+#else 
+                float4 worldPosition : TEXCOORD1;

```

Comment:
For simplicity you could just define `worldPosition` as a float4 (or float3 since it looks like TMP only uses the first two components anyway).

If you pick a float4 then you can just swizzle in the `PointVs...` methods. For example:

`PointVsPlane(IN.worldPosition.xyz, ...`

Reply:
This should be solved with the last push. I removed the #if statements and kept worldPosition as float3 to simplify this the fastest way.